src/tensorwaves/data/tf_phasespace.py

```diff
@@ -25,21 +28,31 @@ def __init__(
         )
 
     def generate(
-        self, size: int, random_generator: UniformRealNumberGenerator
+        self,
+        size: int,
+        seed_or_rng: Optional[Union[UniformRealNumberGenerator, int]] = None,
     ) -> np.ndarray:
-        weights, particles = self.phsp_gen.generate(n_events=size)
+        seed = None
+        if isinstance(seed_or_rng, UniformRealNumberGenerator):
+            seed = seed_or_rng.seed
+        elif isinstance(seed_or_rng, int):
+            seed = seed_or_rng
+        else:
+            raise NotImplementedError(
+                f"Cannot work with a seed or RNG of type {seed_or_rng.__class__.__name__}"
+            )
+        weights, particles = self.phsp_gen.generate(n_events=size, seed=seed)
```

Comment:
I'm a bit puzzled why the seed is given to the generate call. Since this is called several times due to the bunching, every bunch is created with the same seed